data_model/src/transaction.rs

```diff
@@ -667,232 +403,69 @@ impl Ord for TransactionQueryResult {
-    type HashOf = VersionedSignedTransaction;
-
-    #[inline]
-    fn payload(&self) -> &TransactionPayload {
-        &self.as_v1().payload
-    }
-}
-
-impl ValidTransaction {
-    /// Return signatures
-    pub fn signatures(&self) -> impl ExactSizeIterator<Item = &SignatureOf<TransactionPayload>> {
-        self.signatures.iter()
-    }
-}
-
-impl Transaction for ValidTransaction {
-    type HashOf = SignedTransaction;
-
-    #[inline]
-    fn payload(&self) -> &TransactionPayload {
-        &self.payload
-    }
-}
-
-#[cfg(any(feature = "ffi_import", feature = "ffi_export"))]
-declare_versioned!(VersionedRejectedTransaction 1..2, Debug, Clone, PartialEq, Eq, Hash, FromVariant, iroha_ffi::FfiType, IntoSchema);
-#[cfg(all(not(feature = "ffi_import"), not(feature = "ffi_export")))]
-declare_versioned!(VersionedRejectedTransaction 1..2, Debug, Clone, PartialEq, Eq, Hash, FromVariant, IntoSchema);
-
-impl VersionedRejectedTransaction {
-    /// Convert from `&VersionedRejectedTransaction` to V1 reference
-    #[inline]
-    pub const fn as_v1(&self) -> &RejectedTransaction {
-        match self {
-            Self::V1(v1) => v1,
-        }
-    }
-
-    /// Convert from `&mut VersionedRejectedTransaction` to V1 mutable reference
-    #[inline]
-    pub fn as_mut_v1(&mut self) -> &mut RejectedTransaction {
-        match self {
-            Self::V1(v1) => v1,
-        }
-    }
+mod candidate {
+    use parity_scale_codec::Input;
 
-    /// Performs the conversion from `VersionedRejectedTransaction` to V1
-    #[inline]
-    pub fn into_v1(self) -> RejectedTransaction {
-        match self {
-            Self::V1(v1) => v1,
-        }
-    }
-}
-
-impl Transaction for VersionedRejectedTransaction {
-    type HashOf = VersionedSignedTransaction;
-
-    #[inline]
-    fn payload(&self) -> &TransactionPayload {
-        match self {
-            Self::V1(v1) => &v1.payload,
-        }
-    }
-}
-
-impl RejectedTransaction {
-    /// Return signatures
-    pub fn signatures(&self) -> impl ExactSizeIterator<Item = &SignatureOf<TransactionPayload>> {
-        self.signatures.iter()
-    }
-}
-
-impl Transaction for RejectedTransaction {
-    type HashOf = SignedTransaction;
+    use super::*;
 
-    #[inline]
-    fn payload(&self) -> &TransactionPayload {
-        &self.payload
+    #[derive(Decode, Deserialize)]
+    struct SignedTransactionCandidate {
+        payload: TransactionPayload,
+        signatures: SignaturesOf<TransactionPayload>,
     }
-}
 
-impl From<VersionedRejectedTransaction> for VersionedSignedTransaction {
-    fn from(transaction: VersionedRejectedTransaction) -> Self {
-        match transaction {
-            VersionedRejectedTransaction::V1(transaction) => SignedTransaction {
-                payload: transaction.payload,
-                signatures: transaction.signatures,
+    impl SignedTransactionCandidate {
+        fn validate(mut self) -> Result<SignedTransaction, &'static str> {
```

Comment:
Why `&'static str` as error?

Reply:
Because it's only for deserialize/decode. And they work with str